src/main/kotlin/ecommerce/service/AuthService.kt

```diff
@@ -0,0 +1,71 @@
+package ecommerce.service
+
+import ecommerce.auth.JwtTokenProvider
+import ecommerce.dto.AuthResponse
+import ecommerce.dto.LoginForm
+import ecommerce.dto.RegisterForm
+import ecommerce.exception.AuthorizationException
+import ecommerce.exception.InternalServerErrorException
+import ecommerce.exception.MemberEmailAlreadyExistsException
+import ecommerce.model.Cart
+import ecommerce.model.Member
+import ecommerce.repository.CartRepository
+import ecommerce.repository.MemberRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class AuthService(
+    private val memberRepository: MemberRepository,
+    private val cartRepository: CartRepository,
+    private val jwtTokenProvider: JwtTokenProvider,
+) {
+    fun registerMember(form: RegisterForm): Member {
+        checkMemberEmailExists(form.email)
+        val member = Member.from(form)
+        val savedMember = memberRepository.save(member)
+        savedMember.cart = cartRepository.save(Cart())
+        return memberRepository.findByIdOrNull(savedMember.id)
+            ?: throw InternalServerErrorException(MESSAGE_MEMBER_NOT_FOUND)
+    }
+
+    fun loginMember(form: LoginForm): AuthResponse {
+        val member = memberRepository.findByEmail(form.email) ?: throw AuthorizationException(MESSAGE_INVALID_EMAIL)
+        if (member.password != form.password) throw AuthorizationException(MESSAGE_INVALID_PASSWORD)
+        val accessToken = jwtTokenProvider.createToken(member.email)
+        return AuthResponse(accessToken)
+    }
+
+    fun findMemberById(id: Long): Member? = memberRepository.findByIdOrNull(id)
+
+    fun findMemberByEmail(email: String): Member? =
+        memberRepository.findByEmail(email) ?: throw AuthorizationException(MESSAGE_INVALID_EMAIL)
+
+    fun findMemberByToken(token: String): Member {
+        if (!jwtTokenProvider.validateToken(token)) {
+            throw AuthorizationException(MESSAGE_INVALID_TOKEN)
+        }
+        val email = jwtTokenProvider.getPayload(token)
+        val member = memberRepository.findByEmail(email) ?: throw AuthorizationException(MESSAGE_INVALID_EMAIL)
+        return member
```

Comment:
i think jwtTokenProvider can throw exception in itself! 😄 

```suggestion
    fun findMemberByToken(token: String): Member {
        jwtTokenProvider.validateToken(token)
        val email = jwtTokenProvider.getPayload(token)
        return memberRepository.findByEmail(email) ?: throw AuthorizationException(MESSAGE_INVALID_EMAIL)
    }
```

Reply:
Okay, that make sense. 
I will let 'jwtTokenProvider.validateToken()` to throw its exceptions, and introduce new exception handler in global exception handler.